vassal-app/src/main/java/VASSAL/counters/Translate.java

```diff
@@ -193,17 +193,20 @@ protected Command newTranslate(KeyStroke stroke) {
 
     // Mat Support: if we're about to move a Mat, establish the initial relative positions of all its "contents"
     final GamePiece outer = getOutermost(target);
-    if (GameModule.getGameModule().isMatSupport() && !(target instanceof Stack) && !"".equals(outer.getProperty(Mat.MAT_NAME))) {
-      mat = (Mat) Decorator.getDecorator(outer, Mat.class);
-      if (mat != null) {
-        //BR// Should we ONLY take Cargo that are currently selected (in the KeyBuffer)?
-        contents = new ArrayList<>(mat.getContents());
-        offsets = new ArrayList<>();
-        for (final GamePiece piece : contents) {
-          final Point pt = piece.getPosition();
-          pt.x -= p.x;
-          pt.y -= p.y;
-          offsets.add(pt);
+    if (GameModule.getGameModule().isMatSupport() && !(target instanceof Stack)) {
+      final String matName = (String)outer.getProperty(Mat.MAT_NAME);
+      if (!"".equals(matName)) {
+        mat = (Mat) Decorator.getDecorator(outer, Mat.class);
+        if (mat != null) {
+          //BR// Should we ONLY take Cargo that are currently selected (in the KeyBuffer)?
+          contents = new ArrayList<>(mat.getContents());
+          offsets = new ArrayList<>();
+          for (final GamePiece piece : contents) {
+            final Point pt = piece.getPosition();
+            pt.x -= p.x;
+            pt.y -= p.y;
+            offsets.add(pt);
+          }
```

Comment:
Is there someplace this could go so that it's not repeated?

Reply:
If I were to pass those ArrayLists as parameters into a method in e.g. Mat.java, and it filled them up, then upon return would the lists have been updated? So I could call e.g. Mat.makeOffsets(x, y, contents, offsets) and it would fill up "contents" and "offsets" for my later use?

Reply:
Yes, because everything that's not a primitive in Java is a pointer.